fastlane_core/spec/feature_spec.rb

```diff
@@ -0,0 +1,137 @@
+class FeatureHelper
+  def self.disabled_class_method
+  end
+
+  def self.enabled_class_method
+  end
+
+  def disabled_instance_method
+  end
+
+  def enabled_instance_method
+  end
+
+  def self.reset_test_method
+    FeatureHelper.instance_eval { undef test_method }
+  end
+
+  def reset_test_method
+    undef test_method
+  end
+end
+
+describe FastlaneCore do
+  describe FastlaneCore::Feature do
+    describe "Register a Feature" do
+      it "registers a feature successfully with environment variable and description" do
+        expect do
+          FastlaneCore::Feature.register(env_var: "TEST_ENV_VAR", description: "Test environment variable")
+        end.not_to raise_error
+      end
+
+      it "raises an error if no environment variable specified" do
+        expect do
+          FastlaneCore::Feature.register(description: "Test environment variable")
+        end.to raise_error "Invalid Feature"
+      end
+
+      it "raises an error if no description specified" do
+        expect do
+          FastlaneCore::Feature.register(env_var: "TEST_ENV_VAR")
+        end.to raise_error "Invalid Feature"
+      end
+    end
+
+    describe '#enabled?' do
+      after do
+        ENV.delete('TEST_ENV_VAR')
+        FastlaneCore::Feature.features.clear
+      end
+
+      it "reports unregistered features as not enabled" do
+        expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
+      end
+
+      it "reports undefined features as not enabled, even if the environment variable is set" do
+          ENV['TEST_ENV_VAR'] = '1'
+          expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
+      end
+
+      it "reports features for missing environment variables as disabled" do
+        FastlaneCore::Feature.register(env_var: 'TEST_ENV_VAR', description: 'Test environment variable')
+        expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
+      end
+
+      it "reports features for disabled environment variables as disabled" do
+        FastlaneCore::Feature.register(env_var: 'TEST_ENV_VAR', description: 'Test environment variable')
+        ENV['TEST_ENV_VAR'] = '0'
+        expect(FastlaneCore::Feature.enabled?('TEST_ENV_VAR')).to be_falsey
```

Comment:
@mfurtak I believe the rationale here was that it is reasonable for someone to change their env vars to be a string of `'0'` and thinking that they are disabling the feature.